consensus/src/config.rs

```diff
@@ -61,23 +65,15 @@ impl Debug for DelayConfig {
 #[derive(Clone, Debug)]
 pub struct Config {
     /// Identification number of the Member=0,..,(n_members-1).
-    pub node_ix: NodeIndex,
+    pub(crate) node_ix: NodeIndex,
```

Comment:
Augh, I hate `pub(crate)`s with a passion... I think I would prefer even `pub`s, despite the possibility of "evading" the constructor then.

Reply:
But then we kind of miss the point of forcing the check. We could make them private but it would require changing each field access to a getter call.

Reply:
It tells you very little about the interface surface – it's essentially `pub` for the purpose of reasoning about where stuff can be used, but pretends it isn't. We had a lot of instances where someone wanted to meddle in the internals of something in a different part of the code, but didn't want to expose it externally and they would use `pub(crate)` then. In such cases it's much better to either (1) redesign the interface to make more sense for external usage, or (2) restructure the code in such a way that the places that need access to internals get it naturally. Failing that, (3) a more restrictive access grant (e.g. `pub(super)`) is acceptable.

In this case I looked through the code, and unfortunately we use `Config` in quite a couple of places, sometimes quite deep, always assuming the fields are `pub`. Either this should be refactored (solution 2; probably too much work for this PR, but good to keep in mind for future refactors) or they all need to keep the access (solution 3 is not feasible :/). I concede that this is probably the most justified usecase for `pub(crate)` I have seen, so if you really want to ensure nobody creates a config manually you can leave it as is. ._.